src/endpoint/s3/s3_utils.js

```diff
@@ -785,6 +785,15 @@ function key_marker_to_cont_tok(key_marker, objects_arr, is_truncated) {
     return Buffer.from(j).toString('base64');
 }
 
+function invalid_schema_to_aws_error(error) {
+    if (["INVALID_SCHEMA", "INVALID_SCHEMA_PARAMS"].includes(error.rpc_code)) {
+        const err = new S3Error(S3Error.MalformedPolicy);
+        err.message = "Policy was not well-formed or did not validate against the published schema";
```

Comment:
Why do you need to edit the message? you already edited the message property in the error

Reply:
Updated, Thanks